lib/mapper.js

```diff
@@ -160,6 +161,14 @@ function common(facade){
   var platform = platformFromLibrary(lib) || facade.proxy('context.device.type') || (os && os.toLowerCase()) || lib;
   if (platform) ret.platform = sanitizePlatform(platform);
 
+  var userAgent = facade.userAgent();
+  if (platform === 'Web' && userAgent) {
```

Comment:
This would override the `os_name` if one was provided in `context.os.name`. I think we should use the spec'd context fields first, and fallback to fields userAgent second.